lib/solarwinds_apm/patch/tag_sql/sw_mysql2_patch.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require_relative 'annotate_traceparent'
+
+module SolarWindsAPM
+  module Patch
+    module TagSql
+      module SWOMysql2Patch
+        def query(sql, options = {})
+          traceparent = AnnotateTraceparent.generate_traceparent
+          annotated_sql = ''
+
+          if traceparent.empty?
+            annotated_sql = sql
+          else
+            annotated_traceparent = "traceparent='#{AnnotateTraceparent.generate_traceparent}'"
+
+            current_span = ::OpenTelemetry::Trace.current_span
+            attributes_dup = current_span.attributes.dup
+            attributes_dup['sw.query_tag'] = "/*#{annotated_traceparent}*/"
+            current_span.instance_variable_set(:@attributes, attributes_dup.freeze)
```

Comment:
So it's not possible to simply call current_span.add_attribute() here? Could you go a bit into why?

Reply:
Yes, updated